### DISCUSSION TITLE
How to add a port of a board to the official identified micropython boards

### DISCUSSION THREAD

**Discussion by prenou** [micropython/micropython#15278] [Hardware & Peripherals]
Hi I am new in porting micropython for a board.
As I understood for each board which has a port there is a build folder with different files.
My port relates to an M5Stack Atom S3 Lite (esp32s3), so I identified differences between port files for generic esp32 and generic esp32s3, and then I cloned locally all files for the M5Stack Atom which was already defined as a port; I then modified locally the files according the little differences between the 2 boards/MCU.
Which process should I follow to pull my port on official micropython ports?

**Comment by robert-hh** [comment]
The usual way would be to create a directory with the board related files under the board name you chose in a branch of your clone of the master repository. If that all builds and works, you would create a pull request.
If your board is however just a variant of an existing board definition, you could create a board variant. It is mentioned in the board.json file, and the other board files may need additions for dealing with that variant.

In any case you should first check, if a pull request exists for your board. There are at the moment like 50 pull requests for boards outstanding. It seems there is little interest by the maintainer to merge new boards which are very similar to existing ones. That is often the case for ESP32xy and RP2040 based boards. More boards means longer build times for the firmware images, more storage space needed and more maintenance.

**Reply by prenou** [reply]
Hi Robert,
Thank you for that detailled and didactic howto.
I will now investigate my M5Stack AtomS3 Lite (esp32s3) case which here should be easy.
It is for me a preliminary training for later an upcoming 2nd port relating on nRF91 boards.

**Comment by Josverl** [comment]
I'm triggered by your use of the term 'official identified micropython boards' as I have been on the outlook/ a quest to have a simple unique identifier for MicroPython boards for a while , and have not been able to find anything better than : `<port><BOARD>[-VARIANT>]`, where : 

 - `<port>` is the port (in lowercase) , [ stm32 , esp32 , rp2 , samd , ....]
 - `<BOARD>` is the folder in which the board definition is stored , the common norm for this is ALL_CAPS  
 - `<VARIANT>` is used to distinguish two boards that can/should  share the same photo or pins.csv.
   [Different SKUs of the same board are typically variants as well](https://github.com/orgs/micropython/discussions/12294)[(https://github.com/orgs/micropython/discussions/12294). 


so perhaps the board you are after should be a Variant , rather than a new board definition.
If you think it should be a Board , you can also create a separate repo with just the board definition, add MicroPython as a submodule, and compile , and contribut, it that way

**Reply by prenou** [reply]
Hi Josverl,
Thank you for those information which will lead me to achieve clean ports!!!
